java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -163,13 +163,33 @@ private static Tree skipParentheses(Tree parent) {
   }
 
   private static boolean isUnnecessarySubtypeCast(Type childType, TypeCastTree typeCastTree, Type parentType) {
-    boolean isArgument = skipParentheses(typeCastTree.parent()).is(Tree.Kind.ARGUMENTS);
+    Tree parentTree = skipParentheses(typeCastTree.parent());
+    boolean isArgument = parentTree.is(Tree.Kind.ARGUMENTS);
+
     return !childType.isPrimitive()
       // Exception: subtype cast are tolerated in method or constructor call arguments
       && (typeCastTree.type().symbolType().equals(childType)
         || (isArgument && childType.equals(parentType)) || (!isArgument && childType.isSubtypeOf(parentType)))
       && (!ExpressionUtils.skipParentheses(typeCastTree.expression()).is(Tree.Kind.LAMBDA_EXPRESSION)
-        || isUnnecessaryLambdaCast(childType, parentType));
+        || isUnnecessaryLambdaCast(childType, parentType))
+      && !(isArgument && isMandatoryMethodReferenceCast(typeCastTree, parentTree));
+  }
+
+  private static boolean isMandatoryMethodReferenceCast(TypeCastTree typeCastTree, Tree parentTree) {
+    if (typeCastTree.expression().is(Tree.Kind.METHOD_REFERENCE) && skipParentheses(parentTree.parent()).is(Tree.Kind.METHOD_INVOCATION)) {
```

Comment:
As a general observation, I tend to prefer to create an intermediate variable to store `typeCastTree.expression()`.
In my opinion, it increases the clarity of the code and avoids problems in case it returns differents values between two calls, or if the computation is expensive. 
Same for `parentTree.parent()`, to avoid having to call `skipParentheses()` twice.